src/main/java/com/streamr/client/dataunion/DataUnionClient.java

```diff
@@ -0,0 +1,320 @@
+
+    public void setBridgePollTimeout(long bridgePollTimeout){
+        this.bridgePollTimeout = bridgePollTimeout;
+    }
+
+    protected ContractGasProvider mainnetGasProvider() {
+        return new EstimatedGasProvider(mainnet);
+    }
+
+    protected ContractGasProvider sidechainGasProvider() {
+        return new EstimatedGasProvider(sidechain);
+    }
+
+    public DataUnion deployNewDataUnion(String name, String admin, double adminFeeFraction, List<String> agents) throws Exception {
+        if(adminFeeFraction < 0 || adminFeeFraction > 1)
+            throw new NumberFormatException("adminFeeFraction must be between 0 and 1");
+        return deployNewDataUnion(name, admin, toWei(adminFeeFraction), agents);
+    }
+
+    public DataUnion deployNewDataUnion(String name, String admin, BigInteger adminFeeFractionWei, List<String> agents) throws Exception {
+        ArrayList<Address> agentAddresses = new ArrayList<Address>(agents.size());
+        for (String agent : agents) {
+            agentAddresses.add(new Address(agent));
+        }
+        Utf8String duname = new Utf8String(name);
+        factoryMainnet().deployNewDataUnion(
+                new Address(admin),
+                new Uint256(adminFeeFractionWei),
+                new DynamicArray<Address>(Address.class, agentAddresses),
+                duname
+        ).send();
+        Address mainnetAddress = factoryMainnet().mainnetAddress(new Address(mainnetCred.getAddress()), duname).send();
+        String sidechainAddress = factoryMainnet().sidechainAddress(mainnetAddress).send().getValue();
+        return new DataUnion(mainnetDU(mainnetAddress.getValue()), mainnet, mainnetCred, sidechainDU(sidechainAddress), sidechain, sidechainCred);
+    }
+
+    public DataUnion dataUnionFromName(String name) throws Exception {
+        String address = factoryMainnet().mainnetAddress(new Address(mainnetCred.getAddress()), new Utf8String(name)).send().getValue();
+        return dataUnionFromMainnetAddress(address);
+    }
+
+
+    public DataUnion dataUnionFromMainnetAddress(String mainnetAddress) throws Exception {
+        DataUnionMainnet main =  mainnetDU(mainnetAddress);
+        DataUnionSidechain side = sidechainDU(factoryMainnet().sidechainAddress(new Address(main.getContractAddress())).send().getValue());
+        return new DataUnion(main, mainnet, mainnetCred, side, sidechain, sidechainCred);
+    }
+
+    protected DataUnionFactoryMainnet factoryMainnet() {
+        return DataUnionFactoryMainnet.load(mainnetFactory, mainnet, mainnetCred, mainnetGasProvider());
+    }
+
+    protected DataUnionMainnet mainnetDU(String address) {
+        return DataUnionMainnet.load(address, mainnet, mainnetCred, mainnetGasProvider());
+    }
+    protected DataUnionFactorySidechain factorySidechain() {
+        return DataUnionFactorySidechain.load(sidechainFactory, sidechain, sidechainCred, sidechainGasProvider());
+    }
+    protected DataUnionSidechain sidechainDU(String address) {
```

Comment:
Why some methods use word "dataUnion" and others use "DU"?

Reply:
renamed DU -> DataUnion